pkg/nodebootstrap/windows.go

```diff
@@ -13,47 +12,105 @@ import (
 )
 
 type Windows struct {
-	clusterName string
-	ng          *api.NodeGroup
+	clusterConfig *api.ClusterConfig
+	ng            *api.NodeGroup
 }
 
-func NewWindowsBootstrapper(clusterName string, ng *api.NodeGroup) *Windows {
+type keyValue struct {
+	key   string
+	value string
+}
+
+func NewWindowsBootstrapper(clusterConfig *api.ClusterConfig, ng *api.NodeGroup) *Windows {
 	return &Windows{
-		clusterName: clusterName,
-		ng:          ng,
+		clusterConfig: clusterConfig,
+		ng:            ng,
 	}
 }
 
 func (b *Windows) UserData() (string, error) {
-	bootstrapScript := `<powershell>
-[string]$EKSBootstrapScriptFile = "$env:ProgramFiles\Amazon\EKS\Start-EKSBootstrap.ps1"
-`
-	for _, command := range b.ng.PreBootstrapCommands {
-		bootstrapScript += fmt.Sprintf("%s\n", command)
+	bootstrapCommands := []string{
+		`<powershell>
+[string]$EKSBootstrapScriptFile = "$env:ProgramFiles\Amazon\EKS\Start-EKSBootstrap.ps1"`,
 	}
 
-	kubeletOptions := map[string]string{
-		"node-labels":          formatLabels(b.ng.Labels),
-		"register-with-taints": utils.FormatTaints(b.ng.Taints),
+	bootstrapCommands = append(bootstrapCommands, b.ng.PreBootstrapCommands...)
+	eksBootstrapCommand := fmt.Sprintf("& $EKSBootstrapScriptFile %s 3>&1 4>&1 5>&1 6>&1", b.makeBootstrapParams())
+	bootstrapCommands = append(bootstrapCommands,
+		eksBootstrapCommand,
+		"</powershell>",
+	)
+
+	userData := base64.StdEncoding.EncodeToString([]byte(strings.Join(bootstrapCommands, "\n")))
+
+	logger.Debug("user-data = %s", userData)
+	return userData, nil
+}
+
+func (b *Windows) makeBootstrapParams() string {
+	params := []keyValue{
+		{
+			key:   "EKSClusterName",
+			value: b.clusterConfig.Metadata.Name,
+		},
+		{
```

Comment:
are these `b.clusterConfig.X.Y` values always non-nil values? do we need to do any validation to avoid a panic?